cmd/flux/create_source_git.go

```diff
@@ -189,6 +192,13 @@ func createSourceGitCmdRun(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	var ignorePaths *string
+	if len(sourceGitArgs.ignorePaths) > 0 {
+		ignorePathsStr := strings.Join(sourceGitArgs.ignorePaths, "\n")
+		//ignorePathsStr = fmt.Sprintf("%s\n", ignorePathsStr)
+		ignorePaths = &ignorePathsStr
```

Comment:
Right now, if a user does 

```
./bin/flux create source git podinfo \
--url=https://github.com/stefanprodan/podinfo \
--ignore-path=".gitasd/" \
--ignore-path=".github/" \
--export
```

it will result in a manifest that looks like : 

```
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: GitRepository
metadata:
  name: podinfo
  namespace: flux-system
spec:
  ignore: |-
    .gitasd/
    .github/
  interval: 1m0s
  ref:
    branch: master
  url: https://github.com/stefanprodan/podinfo
```

notice the `|-` after the `ignore:` field which means the new line at the end of the multi line string is stripped. if we don't want that, we can add a fake newline at the end of the `ignorePathsStr` after concatenating the list (uncommenting the above line) and then it will result in : 

```
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: GitRepository
metadata:
  name: podinfo
  namespace: flux-system
spec:
  ignore: |
    .gitasd/
    .github/
  interval: 1m0s
  ref:
    branch: master
  url: https://github.com/stefanprodan/podinfo
```
which means no newline chars are stripped. I think this does not make a lot of difference in source-controller and we can leave this as-is (i ll remove the commented out code of course), but i am not very familiar with inner working of source controller. The source controller seems to be using the `ignore` field here - https://github.com/fluxcd/source-controller/blob/55a594a4e13f50f241327015e7e336ad5268b28b/controllers/gitrepository_controller.go#L658 and i think the `ReadPattern` function should work normally. 

Also refer https://yaml-multiline.info/ for an intuitive playground of how `|-` vs `|+` vs `|` work in yaml (i found it handy :))

Reply:
I'm Ok with how is now, removing the last line should cause no issues for source controller.